### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by odeac**
## Overview
Add documentation about how to write test using pytest

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/CORE-1247

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the [developer wiki](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain) for instructions.
- [ ] Your GitHub account is also an account with [Travis CI](https://travis-ci.org). Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: integration-testing/readme.md [hunk 1]
```diff
@@ -54,8 +54,10 @@ Summary: A Python library for the Docker Engine API.
 
 ``` 
 
-# Running the tests
 
+# Test organization
+The tests are organized in a treee structure. The
```
Comment (adaszko) [inline integration-testing/readme.md]: The paragraph looks truncated

File: integration-testing/readme.md [hunk 2]
```diff
@@ -97,5 +107,113 @@ $ ./run_tests.sh --collect-only
...
+For example under `complete_connected` one can find tests that run on a complete connected network.
+
+#### Global Fixtures
+These fixtures are globally defined in `test/conftes.py`
+
+##### config
+This is a fixture that gives access to the command line parameters.
+
+##### docker_network
+This fixture gives access to a docker network with no containers inside
```
Comment (adaszko) [inline integration-testing/readme.md]: These sections should be docstrings in the fixtures' definitions.

File: integration-testing/readme.md [hunk 3]
```diff
@@ -97,5 +107,113 @@ $ ./run_tests.sh --collect-only
...
+used in`conftest.py` and wrapped in a `@pytest.fixture` in order to build a network with the given shape.
+
+This way the setup/teardown logic reused in several places.
+
+### Waiting for conditions
+Because simple `sleep`s are unreliable *all* waiting for various conditions happens.
+
+The wait utilities are defined in `src/tools/wait.py`.
+
+The key function is `wait_for` which waits for a given condition a certain number of seconds. This function checks
```
Comment (adaszko) [inline integration-testing/readme.md]: I have serious objections to this kind of design.  It is much better to have dedicated functions for waiting for particular conditions whose names actually reflect on what they're waiting than a generic `wait_for()` function using untyped callbacks that produces hard to understand stack traces.  It is more important for the test to be readable at a glance more so than to be clever or even short.

**Comment by adaszko** [comment]
For the 'troubleshooting' section of this README:

Problem:
```
(dev)% docker build .
Sending build context to Docker daemon  107.5kB
Step 1/2 : FROM coop.rchain/rnode:latest
Get https://coop.rchain/v2/: Service Unavailable
```

Solution:
```
sbt -mem 5000 clean rholang/bnfc:clean rholang/bnfc:generate compile node/docker:publishLocal
```

File: integration-testing/readme.md (hunk 3 above)
Comment (odeac) [reply integration-testing/readme.md]: This was settled in the other PR so I mark this as resolved

File: integration-testing/readme.md [hunk 4]
```diff
@@ -94,8 +108,141 @@ Examples
...
+def test_network_convergence_a_w(a, w):
+    #...
+def test_network_convergence_a_e(a, e):
+    #...
+def test_network_convergence_b_q(b , q):
+    #...
+...
+```
+
+The difference between the two approaches is the fact that wih `@parametrize.cartezian`, if the fixtures are session,
```
Comment (adaszko) [inline integration-testing/readme.md]: Is it `cartezian` or `cartesian`?  It is spelled `cartesian` below.

**Pull request merged**